cli/spec.py

```diff
@@ -0,0 +1,260 @@
+import ConfigParser
+import os
+import yaml
+import yamlordereddictloader
+
+import clg
+from cli import exceptions
+from cli import utils
+
+SPEC_EXTENSION = '.spec'
+LOOKUP_DICT = {
+    '__DEFAULT__': "default"
+}
+
+TRIM_PARAMS = ['default', 'required', 'requires_only']
+
+
+def cfg_file_to_dict(value):
+    """
+    The custom type for clg spec
+    :param value: the argument value.
+    :return: dict based on a provided ini file.
+    """
+    config = ConfigParser.ConfigParser()
+    with open(value) as fd:
+        config.readfp(fd)
+
+    res_dict = {}
+    for section in config.sections():
+        res_dict[section] = {}
+        for option, value in config.items(section):
+            value = utils.string_to_list(value, append_to_list=False)
+            res_dict[section][option] = value
+
+        res_dict[section].pop('__name__', None)
+
+    return res_dict
+
+
+def parse_args(module_name, config, args=None):
+    """
+    Looks for all the specs for specified module
+    and parses the commandline input arguments accordingly.
+
+    :param module_name: the module name: installer|provisioner|tester
+    :param config: the infrared configuration file
+    :param args: additional arguments to pass to the clg.
+    """
+
+    res_args, subcommand_opts = _get_command_line_args(args, config,
+                                                       module_name)
+
+    # get the default values
+    command_args = subcommand_opts[res_args['command0']]
+    defaults = {k: v.get('default', None)
+                for k, v in command_args.iteritems()}
+
+    # generate config file if required
+    if res_args.get('generate-conf-file', None):
```

Comment:
I see you've removed the call to sys.exit, great!
Do we need to continue with the flow after this if if the user provide the '--generate-conf-file' flag?
If not, you need to break the flow in one of two ways:
1. 'return' in the end of the if block (@tkammer will not like it...)
2. insert the rest of the code into an 'else' block